pkg/controllers/vdb/restart_reconcile.go

```diff
@@ -646,3 +655,16 @@ func (r *RestartReconciler) shouldRequeueIfPodsNotRunning() bool {
 	}
 	return false
 }
+
+// acceptEulaIfMissing will accept the end user license agreement if any pods have not yet signed it
+func (r *RestartReconciler) acceptEulaIfMissing(ctx context.Context) error {
+	for _, p := range r.PFacts.Detail {
+		if !p.eulaAccepted.IsFalse() || !p.isPodRunning {
+			continue
+		}
+		if err := acceptEulaInPod(ctx, p, r.PRunner); err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
I think this is largely duplicated from install_reconciler.go. I am guessing that you couldn't use the one in that file because we couldn't change the function parameters. I suggest moving this logic to a new file (eula.go), and have this file call it and install_reconciler.go. The caller from install_reconciler.go might need a wrapper to call it. Something like this:
```
func (d *InstallReconciler) acceptEulaIfMissing(ctx context.Context) error {
        return acceptEulaIfMissing(ctx, d.PFacts)
}
```

Reply:
Thanks, I like this suggestion.